src/dra/75F54.c

```diff
@@ -760,7 +760,94 @@ void ControlBatForm(void) {
     g_BatScreechDone = screechDone;
 }
 
-INCLUDE_ASM("dra/nonmatchings/75F54", func_801177A0);
+void func_801177A0(void) {
+    byte pad[0x28];
+    s32 i;
+    s32 else_cycles;
+
+    PLAYER.unk19 = 4;
+    func_8010E1EC(0x2000);
+    if (g_Player.pl_vram_flag & 3) {
+        PLAYER.velocityY = 0;
+    }
+    func_8010E234(0x2000);
+    func_8011690C(0);
+    else_cycles = 0;
+    switch (PLAYER.step_s) {
+    case 0:
+        for (i = 0; i < 4; i++) {
+            if (D_800ACED0.pairs[i].unk2 < D_800ACE90[i]) {
+                D_800ACED0.pairs[i].unk2++;
+            } else {
+                else_cycles++;
+            }
+            if (D_800ACE88[i] < D_800ACEC0[i].unk2) {
+                D_800ACEC0[i].unk2--;
+            } else {
+                else_cycles++;
+            }
+            // This means the lower parts only run once!
+            if (i != 0) {
+                continue;
+            }
+            if (g_Player.unk68 != 0) {
+                if (D_8013AECC >= 12) {
+                    continue;
+                }
+                D_8013AECC++;
+            } else {
+                if (g_Player.pl_vram_flag & 0x8000) {
+                    PLAYER.posY.i.hi--;
+                }
+            }
+        }
+
+        if ((g_Player.pl_vram_flag & 3) == 3) {
+            g_Player.unk68 = 1;
+            PLAYER.velocityY = 0;
+            PLAYER.velocityX = 0;
+        }
+        if (else_cycles == 8) {
+            PLAYER.animSet = 1;
+            PLAYER.rotAngle = 0;
+            PLAYER.unk19 = 0;
+            g_Player.unk66 = 1;
+            if (g_Player.unk68 != 0) {
+                PLAYER.step_s = 2;
+                D_800AFDA6[0] = 0xC7;
+            } else {
+                PLAYER.step_s = 1;
+                D_800AFDA6[0] = 0x5F;
+            }
+        }
+        break;
+    case 1:
+        if (g_Player.unk66 == 3) {
+            func_8010E83C(0);
+            PLAYER.posY.i.hi -= 3;
+            if (!(g_Player.pl_vram_flag & 0x8000)) {
+                PLAYER.velocityY = FIX(-1);
+            }
+            PLAYER.palette = 0x8100;
+#if defined(VERSION_US)
+            g_Player.unk20[0] = 0x18;
+#elif defined(VERSION_HD)
+            D_800ACE40 = 0x18;
```

Comment:
I don't have the HD version built so I can't check, but do we know that this variable is outside the `g_Player` struct? This could be checked with the map file or `generated.symbols` file I think.